marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/InputEndpointImpl.java

```diff
@@ -83,12 +92,53 @@ private int getBatchSize() {
 
 		@Override
 		public void accept(InputStream input) {
-			// TODO Auto-generated method stub
+			
+			InputStream output = null;
+			try {
+				queue.put(input);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+			if((queue.size()% getBatchSize()) > 0)
+				return;
+			
+			switch (getPhase()) {
+				case INITIALIZING:
+					setPhase(WorkPhase.RUNNING);
+					break;
+				case RUNNING:
+					break;
+				case INTERRUPTING:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case INTERRUPTED:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				case COMPLETED:
+					throw new MarkLogicInternalException(
+							"cannot accept more input as current phase is  " + getPhase().name());
+				default:
+					throw new MarkLogicInternalException(
+							"unexpected state for " + getEndpointPath() + " during loop: " + getPhase().name());
+			}
+			List<InputStream> inputStreamList = new ArrayList<InputStream>();
+			queue.drainTo(inputStreamList);
+			SessionState session = allowsSession() ? getEndpoint().getCaller().newSessionState() : null;
+			
+			logger.trace("input endpoint running endpoint={} count={} state={}",
+                    getEndpointPath(), getCallCount(), getEndpointState());
+			output = getEndpoint().getCaller().call(
+					getEndpoint().getClient(), getEndpointState(), session, getWorkUnit(), inputStreamList.stream()
+					);
+			
+			if (allowsEndpointState()) {
+                setEndpointState(output);
+            }
```

Comment:
If the indentation were more regular, the code might be easier for someone else to maintain.